ra/ra.go

```diff
@@ -271,31 +271,32 @@ type certificateRequestEvent struct {
 // registration-based overrides are necessary.
 const noRegistrationID = -1
 
-// registrationCounter is a type to abstract the use of
-// ra.SA.CountRegistrationsByIP or ra.SA.CountRegistrationsByIPRange
-type registrationCounter func(context.Context, net.IP, time.Time, time.Time) (int, error)
+// registrationCounter is a type to abstract the use of `CountRegistrationsByIP`
+// or `CountRegistrationsByIPRange` SA methods.
+type registrationCounter func(ctx context.Context, req *sapb.CountRegistrationsByIPRequest) (*sapb.Count, error)
 
 // checkRegistrationIPLimit checks a specific registraton limit by using the
 // provided registrationCounter function to determine if the limit has been
 // exceeded for a given IP or IP range
-func (ra *RegistrationAuthorityImpl) checkRegistrationIPLimit(
-	ctx context.Context,
-	limit ratelimit.RateLimitPolicy,
-	ip net.IP,
-	counter registrationCounter) error {
-
+func (ra *RegistrationAuthorityImpl) checkRegistrationIPLimit(ctx context.Context, limit ratelimit.RateLimitPolicy, ip net.IP, counter registrationCounter) error {
 	if !limit.Enabled() {
 		return nil
 	}
 
 	now := ra.clk.Now()
-	windowBegin := limit.WindowBegin(now)
-	count, err := counter(ctx, ip, windowBegin, now)
+	req := &sapb.CountRegistrationsByIPRequest{
+		Ip: ip,
+		Range: &sapb.Range{
+			Earliest: limit.WindowBegin(now).UnixNano(),
+			Latest:   now.UnixNano(),
+		},
+	}
+	count, err := counter(ctx, req)
```

Comment:
Here (and in most places where you follow the same pattern below) the request object could be built in-line with the gRPC function call:
```go
	count, err := counter(ctx, &sapb.CountRegistrationsByIPRequest{
		Ip: ip,
		Range: &sapb.Range{
			Earliest: limit.WindowBegin(now).UnixNano(),
			Latest:   now.UnixNano(),
		},
	})
```
This is absolutely just a matter of personal preference -- I like the inline style because that way we don't have an extra `req` local variable floating around, but I don't think Boulder has a clear style preference one way or the other overall.